app/src/api.ts

```diff
@@ -604,8 +610,30 @@ export const apiSlice = createApi({
           "source",
           JSON.stringify({
             query: {
-              match: {
-                Name: searchString,
+              bool: {
+                must: [
+                  {
+                    match: {
+                      Name: searchParams.SearchString,
+                    },
+                  },
+                  {
+                    bool: {
+                      must: [
+                        {
+                          terms: {
+                            PriceRange: searchParams.PriceRange,
+                          },
+                        },
+                        {
+                          terms: {
+                            ["Cuisine Types"]: searchParams.CuisineType,
```

Comment:
This is very picky but you can simplify the syntax for the key. Great job on getting the filters working.